app/javascript/react/components/MapButtons/MapButtons.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useRef, useState } from "react";
+import { useTranslation } from "react-i18next";
+
+import clockIcon from "../../assets/icons/clockIcon.svg";
+import copyLinkIcon from "../../assets/icons/copyLinkIcon.svg";
+import filterIcon from "../../assets/icons/filterIcon.svg";
+import shareIcon from "../../assets/icons/shareIcon.svg";
+import { ConfirmationMessage } from "../Modals/atoms/ConfirmationMessage";
+import { CopyLinkModal, CopyLinkModalData } from "../Modals/CopyLinkModal";
+import * as PopupStyles from "../Modals/SessionDetailsModal/SessionDetailsModal.style";
+import { CopyLinkPopup } from "../Modals/SessionDetailsModal/SessionInfo/ModalDesktopHeader";
+import { MapButton } from "./MapButton";
+import * as S from "./MapButtons.style";
+
+enum ButtonTypes {
+  FILTER = "filter",
+  TIMELAPSE = "timelapse",
+  COPY_LINK = "copyLink",
+  SHARE = "share",
+}
+
+const MapButtons = () => {
+  const buttonRef = useRef<HTMLDivElement>(null);
+  const [activeButton, setActiveButton] = useState<ButtonTypes | null>(null);
+  const [buttonPosition, setButtonPosition] = useState({ top: 0, left: 0 });
+  const [showConfirmation, setShowConfirmation] = useState(false);
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    updateButtonPosition();
+    window.addEventListener("resize", updateButtonPosition);
+
+    return () => {
+      window.removeEventListener("resize", updateButtonPosition);
+    };
+  }, [buttonRef.current]);
+
+  const handleClick = (buttonType: ButtonTypes) => {
+    setActiveButton((prevState) =>
+      prevState === buttonType ? null : buttonType
+    );
+  };
+
+  const handleCopyError = (error: Error) => {
+    console.error("Error copying link: ", error.message);
+    alert(t("alert.linkShortenedFailed"));
+  };
+
+  const handleCopySubmit = (
+    formData: CopyLinkModalData,
+    close: { (): void; (): void }
+  ) => {
+    close();
+    setActiveButton(null);
+    setShowConfirmation(true);
+  };
+
+  const updateButtonPosition = () => {
+    if (buttonRef.current) {
+      const rect = buttonRef.current.getBoundingClientRect();
+      setButtonPosition({ top: rect.top, left: rect.left });
+    }
+  };
+
```

Comment:
I'd close confirmation popup after 3s, as we have in modal.

```
useEffect(() => {
    if (showConfirmation) {
      const timer = setTimeout(() => setShowConfirmation(false), 3000);
      return () => clearTimeout(timer);
    }
  }, [showConfirmation]);
```

Reply:
Good catch! Done 🙌